python/metashape_workflow_functions.py

```diff
@@ -1255,3 +1266,21 @@ def finish_run(self):
             file.write("### END CONFIGURATION ###\n")
 
         return True
+
+    def get_paths_written(self):
+        return self.written_paths
+
+    def dump_paths_to_json(self, output_file=None):
+        """
+        Dump the written paths to a JSON file.
+        
+        Args:
+            output_file (str, optional): Path to save the JSON file. If None, will save in output_path with run_id.
+        """
+        if output_file is None:
+            output_file = os.path.join(self.cfg["output_path"], self.run_id + "_paths.json")
+        
+        with open(output_file, 'w') as f:
```

Comment:
I'd recommend ensuring the containing folder exists before trying to save a file. I generally like using [pathlib.mkdir](https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir) but there are equivalents in the `os` module and other places as well.